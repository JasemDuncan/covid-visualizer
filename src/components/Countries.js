import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getCountries } from '../redux/country/countries';

const Countries = () => {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countriesReducer);
  const [searchTerm, setSearchTerm] = useState('');
  useEffect(() => {
    if (countries.length === 0) dispatch(getCountries());
  }, []);

  return (
    <div>
      <div className="row">
        <div className="col">
          <input
            placeholder="search"
            type="text"
            onChange={(event) => {
              setSearchTerm(event.target.value);
            }}
          />
        </div>
      </div>
      <div className="row row-cols-2 row-cols-lg-5 g-2 g-lg-3">

        {
                countries
                  .filter((item) => item.id.toLowerCase().startsWith(searchTerm.toLowerCase()))
                  .map((item) => {
                    const path = item.id;
                    // console.log('ITEMM');
                    // console.log(item);
                    return (
                      <Link className="col" key={item.id} to={`/${path.includes(' ') ? path.replace(/\s/g, '-') : path}`}>
                        <h6>{item.id}</h6>
                        <h6>{item.confirmed}</h6>
                      </Link>
                    );
                  })
          }

      </div>
    </div>
  );
};

export default Countries;
