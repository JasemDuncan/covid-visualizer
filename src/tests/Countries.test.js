import React from 'react';
import '@testing-library/jest-dom';
import {
  render, fireEvent, screen, cleanup, getByTestId,
} from '@testing-library/react';
// import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
// import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Countries from '../components/Countries';
import data from './__mock__/data_mock.json';
// import covid from './__mock__/covid_mock.json';
import store from '../redux/configurationStore';

afterEach(() => {
  cleanup();
});

describe('Country test', () => {
  test('Testing Country component', () => {
    const country = renderer.create(<Provider store={store}><Countries /></Provider>).toJSON();
    expect(country).toMatchSnapshot();
  });

  test('reders content', () => {
    const screen = renderer.create(
      <Provider store={store}>
        <Countries />
      </Provider>,
    ).toJSON();

    expect(screen.children).not.toBeNull();
  });

  // test('count contries',()=>{
  //     const screen = renderer.create(
  //         <Provider store={store}>
  //             <Countries />
  //         </Provider>
  //       ).toJSON();
  //     const countryList=screen.container;

  //     expect(countryList).toBe(countryList);
  // });

  // test('reders content',()=>{
  //    const tree = renderer.create(<Countries />).toJSON();
  //    expect(tree).toMatchSnapshot();
  // });
  // test('reders content',()=>{
  //     const component = render(
  //     <Provider >
  //         <Countries />
  //     </Provider>

  //     );
  //     // console.log(component);
  //     component.getByText('Algeria');

  // });
  // const initialState = {data}
  // const mockStore = configureStore();
  // const store = mockStore(initialState);

  // it('Matches the snapshot', ()=>{
  //     const tree=renderer.create(<Countries />).toJSON();
  //     expect(tree).toMatchSnapshot();
  // });

  // test('Renders correctly', () =>{
  //     const countryComponent = render (
  //         <Provider store={store}>
  //             <Router>
  //                 <Countries />
  //             </Router>
  //         </Provider>,
  //     );
  //     expect(countryComponent).toMatchInlineSnapshot();
  // });

  // test('',()=>{
  //     const countryComponent = render (
  //         <Provider store={store}>
  //             <Router>
  //                 <Countries />
  //             </Router>
  //         </Provider>,
  //     );
  //     const stateList=homeComponent.container.querySelector('h1');
  //     expect(stateList.children).toHaveLength(224);
  // });

  // test('',()=>{
  //     const countryComponent=render(
  //         <Provider store={store}>
  //             <Router>
  //                 <Countries />
  //             </Router>
  //         </Provider>
  //     );
  //     const countryList=countryComponent.container.querySelector('h5');
  //     const searchBar=homeComponent.getByPlaceholderText('search');
  //     fireEvent.change(searchBar, {target: {value: 'pe'}});
  //     expect(countryList.children).toHaveLength(1);

  // });
});
