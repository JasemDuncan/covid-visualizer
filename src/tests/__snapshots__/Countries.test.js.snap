// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country test Testing Country component 1`] = `
<div>
  <div>
    <input
      onChange={[Function]}
      placeholder="search"
      type="text"
    />
  </div>
</div>
`;
